app/models/ex_link.rb

```diff
@@ -0,0 +1,61 @@
+class ExLink < ActiveRecord::Base
+  validates :label, :url, presence: true, uniqueness: true
+  has_many :ex_link_tags
+  has_many :tags, through: :ex_link_tags
+
+  attr_accessor :tagstring
+
+  before_save :add_tags
+
+  has_attached_file :image,
+                    styles: { medium: '300x300>', thumb: '100x100>' },
+                    path: ':rails_root/public/system/images/ex_link/:id/:style/:filename',
+                    url: '/system/images/ex_link/:id/:style/:filename',
+                    default_url: 'img/ex_link/:style/external_link_sample.png'
+  validates_attachment_content_type :image, content_type: /\Aimage\/.*\Z/
+
+  # for nice formating in views
+  def get_tagnames
```

Comment:
`tags.select(:tagname).order(:tagname).uniq.pluck(:tagname)` lets the DB do all the hard work 😄